src/metabase/search/postgres/ingestion.clj

```diff
@@ -27,38 +29,74 @@
        (str/join " ")))
 
 (defn- display-data [m]
-  (select-keys m [:name :display_name :description]))
+  (select-keys m [:name :display_name :description :collection_name]))
 
 (defn- ->entry [m]
   (-> m
       (select-keys
-       [:id
-        :model
-        :archived
-        :collection_id
-        :database_id
-        :table_id])
+       (into [:id :model] search.spec/attr-columns))
       (update :archived boolean)
       (assoc
-       :display_data    (display-data m)
-       :legacy_input    m
+       :display_data (display-data m)
+       :legacy_input m
        :searchable_text (searchable-text m)
-       :model_rank      (model-rank (:model m)))))
+       :model_rank (model-rank (:model m)))))
+
+(defn- legacy-search-items-query
+  "Use an in-place search to get all the items we want to index. A *HACK* only used for search-models without a spec."
+  ([]
+   (let [spec-models (keys (methods search.spec/spec))
+         model-names (reduce disj (disj search.config/all-models "indexed-entity") spec-models)]
+     (legacy-search-items-query model-names)))
+  ([model-names]
+   (if (empty? model-names)
+       ;; Legacy search will return a singleton with a nil record, let's rather not get back anything.
+     {:select [:*] :from :report_card :where [:inline [:= 1 2]]}
+     (-> {:search-string      nil
+          :models             model-names
+            ;; we want to see everything
+          :is-superuser?      true
+          :current-user-id    (t2/select-one-pk :model/User :is_superuser true)
+          :current-user-perms #{"/"}
+          :archived?          nil
+            ;; only need this for display data
+          :model-ancestors?   false}
+         search.legacy/full-search-query
+         (dissoc :limit)))))
+
+(defn- attrs->select-items [attrs]
+  (for [[k v] attrs :when v]
+    (let [as (keyword (u/->snake_case_en (name k)))]
+      (if (true? v) as [v as]))))
+
+;; TODO memoize
+(defn- spec-index-query [search-model]
+  (let [spec (search.spec/spec search-model)]
+    (u/remove-nils
+     {:select (search.spec/qualify-columns :this
+                                           (into [(-> spec :attrs :id (or :id))]
+                                                 (concat
+                                                  (:search-terms spec)
+                                                  (mapcat (fn [k] (attrs->select-items (get spec k)))
```

Comment:
we need to `dissoc` :id from `:attrs` here right?

Reply:
You're right that it'll be be redundant, but it'll also be consistent (when it's present), so it's not a bug.

Not sure it's worth complicating the code, but will take a look.

Reply:
Found a nicer solution - merge a "default attrs" value into the spec when we define it, thanks for pointing out the smell.